runtime-sdk/src/modules/access/test.rs

```diff
@@ -0,0 +1,142 @@
+use super::{Module as Access, Error as AccessError, types::{Authorization, MethodAuthorization}};
+
+struct TestConfig;
+
+impl core::Config for TestConfig {}
+
+impl modules::access::Config for TestConfig {
+    const METHOD_AUTHORIZATIONS: Lazy<Authorization> = Lazy::new(|| {
+        Authorization::with_filtered_methods([
+            (
+                "test.FilteredMethod",
+                MethodAuthorization::allow_from([
+                    keys::alice::address(),
+                ]),
+            ),
+        ])
+    });
+}
+
+/// Test runtime.
+struct TestRuntime;
+
+impl Runtime for TestRuntime {
+    const VERSION: Version = Version::new(0, 0, 0);
+
+    type Core = modules::core::Module<TestConfig>;
+
+    type Modules = (modules::core::Module<TestConfig>, modules::access::Module<TestConfig>, TestModule);
+
+    fn genesis_state() -> <Self::Modules as module::MigrationHandler>::Genesis {
+        (
+            core::Genesis {
+                parameters: core::Parameters {
+                    max_batch_gas: 10_000_000,
+                    min_gas_price: BTreeMap::from([(Denomination::NATIVE, 0)]),
+                    ..Default::default()
+                },
+            },
+            (), // Access module has no genesis.
+            (), // Test module has no genesis.
+        )
+    }
+}
+
+/// A module with multiple no-op methods; intended for testing routing.
+struct TestModule;
+
+#[sdk_derive(Module)]
+impl TestModule {
+    const NAME: &'static str = "test";
+    type Error = core::Error;
+    type Event = ();
+    type Parameters = ();
+    type Genesis = ();
+
+    #[handler(call = "test.FilteredMethod")]
+    fn filtered_method<C: Context>(_ctx: &C, fail: bool) -> Result<(), core::Error> {
+        Err(core::Error::Forbidden)
+    }
+
+    #[handler(call = "test.AllowedMethod")]
+    fn subcall<C: Context>(ctx: &C, _args: ()) -> Result<(), core::Error> {
+        Ok(())
+    }
+}
+
+impl module::BlockHandler for TestModule {}
+impl module::TransactionHandler for TestModule {}
+impl module::InvariantHandler for TestModule {}
+
+fn dispatch_test<C: Context>(ctx: &C, sender: SignatureAddressSpec, meth: &str, should_fail: bool) {
+    let tx = transaction::Transaction {
+        version: 1,
+        call: transaction::Call {
+            format: transaction::CallFormat::Plain,
+            method: meth.to_owned(),
+            body: cbor::to_value(()),
+            ..Default::default()
+        },
+        auth_info: transaction::AuthInfo {
+            signer_info: vec![transaction::SignerInfo::new_sigspec(
+                sender,
+                0,
+            )],
+            fee: transaction::Fee {
+                amount: Default::default(),
+                gas: 1000,
+                consensus_messages: 0,
+            },
+            ..Default::default()
+        },
+    };
+    let call = tx.call.clone();
+
+    // Try to transfer.
+    CurrentState::with_transaction_opts(Options::new().with_tx(tx.into()), || {
```

Comment:
Can we instead go through the dispatcher using the mock signer as this will test the usual dispatch flow. See e.g. [accounts tests](https://github.com/oasisprotocol/oasis-sdk/blob/dfa1d9bbe7a60ac088926a60d97488f83883a071/runtime-sdk/src/modules/accounts/test.rs#L1287) for details.